sql/core/src/test/scala/org/apache/spark/sql/sources/v2/TestInMemoryTableCatalog.scala

```diff
@@ -252,6 +252,32 @@ class InMemoryTable(
       withData(messages.map(_.asInstanceOf[BufferedRows]))
     }
   }
+
+  private def splitAnd(filter: Filter): Seq[Filter] = {
+    filter match {
+      case And(left, right) => splitAnd(left) ++ splitAnd(right)
+      case _ => filter :: Nil
+    }
+  }
+
+  override def deleteWhere(filters: Array[Filter]): Unit = dataMap.synchronized {
+    val deleteKeys = dataMap.keys.filter { partValues =>
+      filters.flatMap(splitAnd).forall {
+        case EqualTo(attr, value) =>
+          partFieldNames.zipWithIndex.find(_._1 == attr) match {
+            case Some((_, partIndex)) =>
+              value == partValues(partIndex)
+            case _ =>
+              throw new IllegalArgumentException(s"Unknown filter attribute: $attr")
+          }
+        case IsNotNull(_) =>
+          true
```

Comment:
Is this always true? What happens if the insert used a `null` partition value?

Reply:
Fixed.